bitcoin/src/p2p/message.rs

```diff
@@ -536,6 +536,12 @@ impl Decodable for RawNetworkMessage {
     }
 }
 
+impl From<RawNetworkMessage> for NetworkMessage {
+    fn from(value: RawNetworkMessage) -> Self {
+        value.payload
+    }
+}
```

Comment:
I'm not saying it's wrong but it looks kinda strange. A more natural approach would be to have `into_parts(self) -> (Magic, NetworkMessage)` method. However maybe this struct should simply have the fields public. And `Raw` suggest it contains just a `Vec<u8>` which is not the case, so it should be named `NetworkMessage` and the other one `NetworkMessagePayload`. Or something like that. (Of course the naming is OT and you don't have to solve it.)

Reply:
I see `master` has a better alternative `into_payload` not present in 0.32.0. Implementing `From` is redundant